core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java

```diff
@@ -0,0 +1,288 @@
+import org.apache.accumulo.core.util.FastFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AccumuloStore<T> extends AbstractFateStore<T> {
+
+  private static Logger log = LoggerFactory.getLogger(AccumuloStore.class);
+
+  private static final Comparator<Entry<Key,Value>> repoComparator =
+      Comparator.comparing(o -> o.getKey().getColumnQualifier(), reverseOrder());
+
+  private final ClientContext context;
+  private final String tableName;
+
+  public AccumuloStore(ClientContext context, String tableName) {
+    this.context = Objects.requireNonNull(context);
+    this.tableName = Objects.requireNonNull(tableName);
+  }
+
+  @Override
+  public long create() {
+    long tid = RANDOM.get().nextLong() & 0x7fffffffffffffffL;
+
+    // TODO = conditional mutation and retry if exists?
+    newMutator(tid).putStatus(TStatus.NEW).putCreateTime(System.currentTimeMillis()).mutate();
+
+    return tid;
+  }
+
+  @Override
+  protected List<String> getTransactions() {
+    return scanTx(scanner -> {
+      scanner.setRange(new Range());
+      scanner.fetchColumn(TxColumnFamily.STATUS_COLUMN.getColumnFamily(),
+          TxColumnFamily.STATUS_COLUMN.getColumnQualifier());
+      return StreamSupport.stream(scanner.spliterator(), false)
+          .map(e -> e.getKey().getRow().toString()).collect(Collectors.toList());
+    });
+  }
+
+  @Override
+  protected TStatus _getStatus(long tid) {
+    return scanTx(scanner -> {
+      scanner.setRange(getRow(tid));
+      scanner.fetchColumn(TxColumnFamily.STATUS_COLUMN.getColumnFamily(),
+          TxColumnFamily.STATUS_COLUMN.getColumnQualifier());
+      return StreamSupport.stream(scanner.spliterator(), false)
+          .map(e -> TStatus.valueOf(e.getValue().toString())).findFirst().orElse(TStatus.UNKNOWN);
+    });
+  }
+
+  @Override
+  protected FateTxStore<T> newFateTxStore(long tid, boolean isReserved) {
+    return new FateTxStoreImpl(tid, isReserved);
+  }
+
+  static Range getRow(long tid) {
+    return new Range("tx_" + FastFormat.toHexString(tid));
+  }
+
+  private <T> FateMutatorImpl<T> newMutator(long tid) {
+    return new FateMutatorImpl<>(context, tableName, tid);
+  }
+
+  private <R> R scanTx(Function<Scanner,R> func) {
+    try (Scanner scanner = context.createScanner(tableName, Authorizations.EMPTY)) {
+      return func.apply(scanner);
+    } catch (TableNotFoundException e) {
+      throw new IllegalStateException(tableName + " not found!", e);
+    }
+  }
+
+  private class FateTxStoreImpl extends AbstractFateTxStoreImpl<T> {
+
+    private FateTxStoreImpl(long tid, boolean isReserved) {
+      super(tid, isReserved);
+    }
+
+    @Override
+    public Repo<T> top() {
+      verifyReserved(false);
+
+      return scanTx(scanner -> {
+        scanner.setRange(getRow(tid));
+        scanner.fetchColumnFamily(RepoColumnFamily.NAME);
+        return StreamSupport.stream(scanner.spliterator(), false).sorted(repoComparator).map(e -> {
```

Comment:
Optimizing getTop() could be a follow on issue.

Reply:
@keith-turner - Thanks for the suggestion on encoding, I was going to ask your thought on this but forgot. As I was working on this I wasn't sure the best approach with the encoding so I just used the comaparator for now.

I was hoping there might be some utility or iterator setting that existed already that may make it simple to reverse the order because reversing the order of sorting seems like a common use case. But the only thing I found when searching in the code was was `ReverseLexicoder` but wasn't sure that made sense to use (or if i even would work here, I haven't looked close enough). Your suggesting is pretty simple so should work fine.

I think we could set the batch size on the scanner for now to be smaller unless we go with the optimization of getTop() later you mentioned

Reply:
>  But the only thing I found when searching in the code was was ReverseLexicoder but wasn't sure that made sense to use (or if i even would work here, I haven't looked close enough).

Using `Lexicoder<Integer> lc = new ReverseLexicoder(new IntegerLexicoder())` would work nicely. It would not be human readable though.  For this case since it has a small number of expected integers, making it human readable and counting down from 99 seems ok.